leap_ec/probe.py

```diff
@@ -328,37 +350,39 @@ def __call__(self, population):
         assert (population is not None)
         assert ('leap' in self.context)
         assert ('generation' in self.context['leap'])
-
-        # Always update the best-so-far variable
-        best_ind = best_of_gen(population)
-        if self.bsf_ind is None or (best_ind > self.bsf_ind):
-            self.bsf_ind = best_ind
-
-        # Check if we've reached a measurement interval
-        generation = self.context['leap']['generation']
-        if generation % self.modulo != 0:
-            # If not, don't write fitness info
-            return population
-        else:
-            # Do write fitness info
-            if self.job is not None:
-                self.stream.write(str(self.job) + ', ')
-            for _, v in self.notes.items():
-                self.stream.write(str(v) + ', ')
-
-            self.stream.write(str(generation) + ', ')
-
-            self.stream.write(str(self.bsf_ind.fitness) + ', ')
-
-            fitnesses = [x.fitness for x in population]
-            self.stream.write(str(np.mean(fitnesses)) + ', ')
-            self.stream.write(str(np.std(fitnesses)) + ', ')
-            self.stream.write(str(np.min(fitnesses)) + ', ')
-            self.stream.write(str(np.max(fitnesses)))
-            for _, f in self.extra_metrics.items():
-                self.stream.write(', ' + str(f(population)))
-            self.stream.write('\n')
-            return population
+        
+        with _maybe_json(self.numpy_as_json):
```

Comment:
Most folks see `with` in the context of opening a file, so this would be a little weird to see. So, again, a "why" comment would be helpful here.